toolbox/library/src/CMM.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include "WBToolbox/Block/CMM.h"
+#include "WBToolbox/Base/Configuration.h"
+#include "WBToolbox/Base/RobotInterface.h"
+
+#include <BlockFactory/Core/BlockInformation.h>
+#include <BlockFactory/Core/Log.h>
+#include <BlockFactory/Core/Parameter.h>
+#include <BlockFactory/Core/Parameters.h>
+#include <BlockFactory/Core/Signal.h>
+#include <Eigen/Core>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/Indices.h>
+
+#include <memory>
+#include <ostream>
+#include <tuple>
+
+using namespace wbt::block;
+using namespace blockfactory::core;
+
+// INDICES: PARAMETERS, INPUTS, OUTPUT
+// ===================================
+
+enum InputIndex
+{
+    BasePose = 0,
+    JointConfiguration,
+};
+
+enum OutputIndex
+{
+    CMM = 0,
+};
+
+// BLOCK PIMPL
+// ===========
+
+class CMM::impl
+{
+public:
+    iDynTree::MatrixDynSize CentroidalTotalMomentumMatrix;
+};
+
+// BLOCK CLASS
+// ===========
+
+CMM::CMM()
+    : pImpl{new impl()}
+{}
+
+CMM::~CMM() = default;
+
+
+
```

Comment:
I don't think that the clang-format of this repo allows multiple blank lines. Are you sure you're using it? If not, can you update the style of all the C++ file with that?

Reply:
Yes sure !

Reply:
Done in commit 95fa7bcb51c3f942271b80777c4861f7eb814e3c